internal/core/command/types.go

```diff
@@ -15,19 +15,29 @@ package command
 
 import (
 	"bytes"
+	"fmt"
+	"net/http"
+
 	"github.com/edgexfoundry/edgex-go/internal"
 	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
-	"net/http"
 )
 
 // DefaultErrorCode the default value used when an error has occurred and no response code was received.
 const DefaultErrorCode = 0
 
 // ErrDeviceLocked error indicating the locked state of a device
-type ErrDeviceLocked struct{}
+type ErrDeviceLocked struct {
+	id   string
+	name string
+}
 
-func (ErrDeviceLocked) Error() string {
-	return "The device is in admin locked state"
+func (e ErrDeviceLocked) Error() string {
```

Comment:
If at any time only id or name is provided, we could just have Error() defined as 
func (e ErrDeviceLocked) Error() string {
         return fmt.Sprintf("device '%s%s' is in admin locked state", e.id, e.name)
}

Reply:
@difince Need to define errors consistently. See [here](https://github.com/edgexfoundry/edgex-go/blob/master/internal/core/metadata/errors/types.go). Specifically, we `New()` them and then `Error()` returns the error message. In this case, the `Error()` implementation can be streamlined as @mkbhanda says.
```
type ErrDeviceLocked struct {
    device string
}

func NewErrDeviceLocked(device string) error {
    return ErrDeviceLocked{device: device}
}

func (e ErrDeviceLocked) Error() string {
    return fmt.Sprintf("device '%s' is in admin locked state", e.id)
}
```
If you were to declare this in an `errors` package similar to how it's done in core-metadata, then the `device` property on the type would not be externally accessible.

Reply:
I have added such struct initiated by New() and now we are able to send in the error msg whatever we want from type string to identify the device - id,  name and etc